### ISSUE TITLE
Problem building docs

### ISSUE THREAD

**Issue by lewisl** [JuliaStats/DataFrames.jl#927]
Sorry for the nube problem, but I was trying to build the API docs per the instructions.   The instructions leave it to the imagination to determine what directory one should be in when running build.jl.

The following errors were produced:
`ERROR: LoadError: UndefVarError: Docs not defined
 in find_loaded_packages at /Users/lewislevin/.julia/v0.4/Docile/src/Cache/types.jl:44
 in update! at /Users/lewislevin/.julia/v0.4/Docile/src/Cache/types.jl:55
 in update! at /Users/lewislevin/.julia/v0.4/Docile/src/Cache/types.jl:67
 in loadedmodules at /Users/lewislevin/.julia/v0.4/Docile/src/Cache/types.jl:76
 in metadata at /Users/lewislevin/.julia/v0.4/Docile/src/Interface/legacy.jl:39
 in include at /Applications/Julia-0.4.3.app/Contents/Resources/julia/lib/julia/sys.dylib
 in include_from_node1 at /Applications/Julia-0.4.3.app/Contents/Resources/julia/lib/julia/sys.dylib
while loading /Users/lewislevin/.julia/v0.4/DataFrames/docs/build.jl, in expression starting on line 62`

I am sure this is super simple and can be fixed by just starting from the right directory.  

Sorry to point out that it really is necessary to be super clear about this--there was no obvious way to determine this.  If you do it daily it's easy--one's first time is a bit of a mystery.

**Comment by LouisMagarshack** [comment]
Trying it out on my side, there seems to be a few issues.

* `Docs` is defined in `Base.Docs` and `Docile.Interface`
* `@doc_str` is defined in `Base.Docs` and `Docile`

So I added:
```Docs = Docile.Interface.Docs                                                                                           
doc_str = getfield(Docile, symbol("@doc_str")) ```
at the top of my build.jl to fix the ambiguity, but then I had an error inside Lexicon...

`ERROR: LoadError: KeyError: source not found`

So I changed Lexicon.jl's Base.filter to check keys before accessing them:
```function Base.filter(docs::Metadata; categories = CATEGORY_ORDER, files = AbstractString[])                            
    result = copy(docs)                                                                                                
    if length(files) > 0                                                                                               
        filter!((k,v) -> any(x -> haskey(v.data, :source) && contains(v.data[:source][2], x), files), result.entries)  
    end                                                                                                                
    if length(categories) > 0                                                                                          
        filter!((k,v) -> any(x -> category(v) == x, categories), result.entries)                                       
    end                                                                                                                
    result                                                                                                             
end```

That gave me the following output:
```INFO: writing documentation to api/maintypes.md
INFO: writing documentation to api/utilities.md
INFO: writing documentation to api/manipulation.md```

Since I don't really know how any of the packages work, I don't know if those changes are breaking or not or how to test any of it.

PS: build.jl calls to `beginswith` instead of `startswith` and that is an easy enough fix...

**Comment by tshort** [comment]
Base docs, Docile, and Lexicon have all changed quite a bit since this was written. I think there's still work progressing on auto-building of docs. Here's an interesting new package:

https://github.com/MichaelHatherly/Lapidary.jl

Anyway, build.jl might need a fair bit of work.

**Comment by tshort** [comment]
Closed by #929.
